### ISSUE TITLE
Lost network connexion in container due to clock_gettime failed ?

### ISSUE THREAD

**Issue by danielp123** [linuxserver/docker-nextcloud#175]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior

I updated the container on my Raspberry Pi 4 to the latest version (Jan16 2021) and Nextcloud stopped working. When I go to the subdomain, I get an error:
Internal Server Error

The server encountered an internal error and was unable to complete your request.
Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.
More details can be found in the server log.

## Current Behavior

I looked at the Nextcloud log and it said it couldn't connect to the database (I use mariadb in another container if that's important). But the mariadb container is up and working; I didn't change anything from previous version (I use docker-compose so all parameters are identical as before).

## Steps to Reproduce

I tried to find out what the problem was inside the container. Here's what I typed:

1. sudo docker exec -it nextcloud /bin/bash
2. nslookup google.com:
nslookup: clock_gettime(MONOTONIC) failed
3. date:
Thu Jan  1 01:00:00 CET 1970

## Environment
**OS: Raspberry Pi OS**
**CPU architecture:** arm32
**How docker service was installed: sudo apt install docker**

## Command used to create docker container (run/create/compose/screenshot)

---
version: "2.1"
services:
  nextcloud:
    image: ghcr.io/linuxserver/nextcloud
    container_name: nextcloud
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Zurich
    volumes:
      - $HOME/appdata/nextcloud/config:/config
      - $HOME/data:/data
    depends_on:
      - mariadb
    restart: unless-stopped
  mariadb:
    image: linuxserver/mariadb
    container_name: mariadb
    environment:
      - PUID=1000
      - PGID=1000
      - MYSQL_ROOT_PASSWORD=***
      - TZ=Europe/Zurich
      - MYSQL_DATABASE=nextcloud
      - MYSQL_USER=***
      - MYSQL_PASSWORD=***
    volumes:
      - $HOME/appdata/mariadb:/config
    restart: unless-stopped
  swag:
    image: linuxserver/swag
    container_name: swag
    cap_add:
      - NET_ADMIN
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Zurich
      - URL=***.com
      - SUBDOMAINS=wildcard
      - VALIDATION=dns
      - DNSPLUGIN=cloudflare
    volumes:
      - $HOME/appdata/swag/config:/config
      - $HOME/data:/data
    ports:
      - 443:443
    restart: unless-stopped

## Docker logs

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-envfile: executing... 
[cont-init.d] 01-envfile: exited 0.
[cont-init.d] 10-adduser: executing... 

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \ 
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/

Brought to you by linuxserver.io
-------------------------------------

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 20-config: executing... 
[cont-init.d] 20-config: exited 0.
[cont-init.d] 30-keygen: executing... 
generating self-signed keys in /config/keys, you can replace these with your own keys if required
Generating a RSA private key
............+++++
...................................+++++
writing new private key to '/config/keys/cert.key'
-----
3069834128:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
[cont-init.d] 30-keygen: exited 1.
[cont-init.d] 40-config: executing... 
[cont-init.d] 40-config: exited 0.
[cont-init.d] 50-install: executing... 
tar: index.php: time stamp 2021-01-14 10:35:08 is 1172020807 s in the future
tar: ocs-provider/index.php: time stamp 2021-01-14 10:35:08 is 1172020807 s in the future
tar: ocs-provider: time stamp 2021-01-14 10:35:08 is 5263001911605069516 s in the future
tar: ocs/v2.php: time stamp 2021-01-14 10:35:08 is 1172020807 s in the future
tar: ocs/providers.php: time stamp 2021-01-14 10:35:08 is 1162021029 s in the future
tar: ocs/v1.php: time stamp 2021-01-14 10:35:08 is 1162021029 s in the future
tar: ocs: time stamp 2021-01-14 10:35:08 is 5263001911605069516 s in the future
tar: resources/update-locales.php: time stamp 2021-01-14 10:35:08 is 1162021029 s in the future
.... lots of files listed that are XXX s in the future ....
ends with:
nginx: [emerg] cannot load certificate "/config/keys/cert.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/config/keys/cert.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)

The last line is really weird because I can definitively see cert.crt in my /config/keys directory.

**Comment by Roxedus** [comment]
You are running into this. https://docs.linuxserver.io/faq#my-host-is-incompatible-with-images-based-on-ubuntu-focal-and-alpine-3-13

**Comment by jdespraz** [comment]
I had the same issue. Thanks for the link @Roxedus, that solved the problem.

If anyone is interested, here's how I did it (the procedure corresponds to option 2 in @Roxedus link):
```
# download a version of libsecomp2 >= 2.4.4 (see https://packages.debian.org/buster-backports/libseccomp2)
wget http://ftp.de.debian.org/debian/pool/main/libs/libseccomp/libseccomp2_2.4.4-1~bpo10+1_armhf.deb
# install the new library (make sure you have libc6>2.17 installed)
sudo dpkg -i libseccomp2_2.4.4-1~bpo10+1_armhf.deb
```
Then restart your docker and Nextcloud is back up and running smoothly!

**Comment by danielp123** [comment]
Thank you @Roxedus and @jdespraz for your help. I followed instructions on the link @Roxedus gave, implemented option 2, but I still have the same issue. Tried option 3 also, but raspberry pi said:
libseccomp2 is already the newest version (2.4.4-1~bpo10+1).

I also made sure I had the libc6>2.17:
`ldd --version`
ldd (Debian GLIBC 2.28-10+rpi1) 2.28

Stopped docker-compose, even reinstalled it, but nothing changed. Where should I look next?

**Comment by j0nnymoe** [comment]
If you're on a rpi4, upgrading to a 64bit OS would be the best solution for this.

**Comment by Robbt** [comment]
So I managed to fix it by installing the 2.4.4 libseccomp2 and upgrading to the latest docker-ce. (It didn't work until doing this)
I also switched my raspberry pi to 64-bit OS see this uhttps://medium.com/for-linux-users/how-to-make-your-raspberry-pi-4-faster-with-a-64-bit-kernel-77028c47d653

I'm not sure if switching to the 64bit OS was necessary or if I just needed to update docker after installing the new libseccomp package. But I got my nextcloud instance back after a few hours of work. (this is why I ignore updates).
